tornado/websocket.py

```diff
@@ -143,6 +148,23 @@ class WebSocketClosedError(WebSocketError):
     pass
 
 
+class WebSocketRedirect(WebSocketError):
+    """Raised when code is 302/303 in received headers
+       while doing websocket handshake
+    """
+
+    def __init__(self, target_location: str) -> None:
+        self.target_location = target_location
+
+
+class RequestRedirectError(WebSocketError):
```

Comment:
Be consistent is always very import for a project. But from my perspective something like SimpleHTTPClient or AsyncSimpleHTTPClient is very different compared to a WebSocketClientConnection, HTTPClient is one level higher than a connection. That's also why a redirect exception is raised here. For a HTTPClient, when redirecting, just do another fetch to target
request is enough. But for a connection, redirect means this connection is going to be closed and we should start a new connection to the redirected location. I cannot find a HTTPError(302/303) in 
simple_httpclient.py. WebSocketRedirectError is what I could find as a simple and clear word to represent what is going to happen. I'm really open to accept any other suitable name

Reply:
`SimpleAsyncHTTPClient.fetch()` and `websocket_connect()` are in the same situation regarding redirects. Since the Simple client does not (currently) support keep-alive connections, another fetch is always another connection. Since `websocket_connect()` uses the Simple client, that's true for it too. But if SimpleAsyncHTTPClient supported keep-alive connections, then it would re-use the same connection for a redirect to another path on the same host:port, and so could `websocket_connect()` do the exact same.